### ISSUE TITLE
Support specify the proxy for GitHub Desktop (not for Git)

### ISSUE THREAD

**Issue by SadPencil** [desktop/desktop#13975]
### Describe the feature or problem you’d like to solve

China blocks github.com. The proxy for performing git operations can be specify at `.gitconfig` file **BUT** there is no way to set the proxy for GitHub Desktop GUI itself.

Then when a user wants to log in to GitHub, GitHub Desktop will require the user open a link in the browser, and let the browser make a callback to GitHub Desktop, then GitHub Desktop will connect to github.com directly without a proxy, and will not successfully finish the login. 

### Proposed solution

Let GitHub Desktop support the proxy **for itself** (not for Git) through at least one of environment variables, command arguments, and GUI settings.

### Additional context

![](https://user-images.githubusercontent.com/11227602/155250522-44226bfb-5ed7-4fff-b5fb-7038a9f38567.png)
![](https://user-images.githubusercontent.com/11227602/155250536-a5d2b5cb-cf4f-4e2c-a0ec-5609757c0cc4.png)

Note: it is different from setting `git config --global http.proxy http://proxy.server.com:8080`.

**Comment by steveward** [comment]
Thanks for the issue @SadPencil.

Unfortunately GitHub Desktop [doesn't support authenticating proxies](https://github.com/desktop/desktop/blob/development/docs/technical/proxies.md#authenticating-proxies) at this time.

Some users have reported success with installing a local proxy which can then deal with the upstream authentication, such as [cntlm](http://cntlm.sourceforge.net) or [winfoom](https://github.com/ecovaci/winfoom) but it's not anything we have experience with.

I'm going to close this in favor of https://github.com/desktop/desktop/issues/10026, please follow along in that issue for updates.

**Comment by SadPencil** [comment]
No, I suggest re-open this issue. @steveward 
This issue is not for git cloneing stuff, it is for the GitHub Desktop GUI itself, i.e. network traffic interacting with github.com instead of git repos. This remainder has been mentioned in this issue twice.
Also, it has nothing to do with authenticating proxies #10026.

**Comment by niik** [comment]
@SadPencil GitHub Desktop should

**Comment by niik** [comment]
@SadPencil GitHub Desktop should use the system-configured proxy for all network operations. What operating system are you running and have you configured a proxy in your system settings?

**Comment by SadPencil** [comment]
> @SadPencil GitHub Desktop should use the system-configured proxy for all network operations. What operating system are you running and have you configured a proxy in your system settings?

Yes. It works if the system-wide proxy is set. Thanks for the tip. It seems that the documentation about GitHub Desktop does not mentioned about this feature.
OS: Windows 11 21H2 amd64 (22000.527)

This issue is a feature-request instead of a bug report, for a customizable proxy option that does not depends on the system-wide proxy settings (in order to reduce the proxy traffic for other apps).

**Comment by steveward** [comment]
@SadPencil I'm glad to hear you were able to get the system proxy working, and apologies for my misunderstanding of your request.

**Comment by aadityaverma** [comment]
I am getting the error while commiting a push on a systemwide socks proxy on windows 10 .. "fatal: unable to access 'https://github.com/aadityaverma/pycrop-yield-prediction.git/': SOCKS4: Failed receiving connect request ack: Failure when receiving data from the peer" ..

How do I force github to use socks5 ?

**Comment by sfc9982** [comment]
What's your proxy software?

**Comment by benbrummer** [comment]
We have a use case for a specific local proxy server, with the only purpuose to connect to a GitHub Enterprise Instance. Configuring it as a system proxy breaks everything else. So having a custom proxy option within GitHub Desktop would be great.
A not feasible workaround ist to configure a proxy server on commandline

`Set-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings" -Name ProxyServer -Value 127.0.0.1:PORT`

And activate/deactivate it for a short time

```
Set-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings" -Name ProxyEnable -Value 1
Set-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings" -Name ProxyEnable -Value 0
```

**Comment by Bratzmeister** [comment]
> We have a use case for a specific local proxy server, with the only purpuose to connect to a GitHub Enterprise Instance. Configuring it as a system proxy breaks everything else. So having a custom proxy option within GitHub Desktop would be great.

we have the same use case and are currently unable to connect to our Github Enterprise instance via GitHub Desktop because the Windows Proxy Settings are ignored when using VPN (which is a requirement for us)
